app/services/train_api.rb

```diff
@@ -9,4 +9,14 @@ def index
     response = self.class.get('/v1/trains')
     JSON.parse(response.body)
   end
+
+  def show(params)
+    response = self.class.get("/v1/trains/#{params[:id]}")
+    JSON.parse(response.body)
+  end
+
+  def update(id, params)
+    train_params = params.require(:train).permit(:name, :train_model_name, :number_of_cars, :max_weight_capacity, :active)
```

Comment:
i wonder if we can permit these params back in the controller and not do it here, like all the other controllers.
then we can just pass train_params to this update action

Reply:
> i wonder if we can permit these params back in the controller and not do it here, like all the other controllers.
> then we can just pass train_params to this update action

`param is missing or the value is empty: train` i get this error in the request specs when i try to do it in the controller instead

Reply:
yep so update your specs

Reply:
getting an error in tests isnt an excuse to not change this, it just means you need to change your tests to suit the new logic